CMake fails to get results from PkgConfig

My solution was to export the proper locations through external variables.
This patch [0000-fix_header_locations.txt](https://github.com/maliit/framework/files/5785552/0000-fix_header_locations.txt)
Or this should work as a temporary fix:

`target_include_directories(inputpanel-shell PUBLIC $ENV{wayland_private_headers_devel_includes})`
`target_include_directories(inputpanel-shell PUBLIC $ENV{xkbcommon_includes})`
`target_include_directories(maliit-common PUBLIC $ENV{xkbcommon_includes})`

Simon from the Plasma Mobile community proposed adding something similar to this at the lists:
`find_package(PkgConfig REQUIRED)` <-- already on the lists
`pkg_check_modules(IMGUI REQUIRED IMPORTED_TARGET GLOBAL imgui)`
`target_link_libraries(OgreOverlay PRIVATE PkgConfig::IMGUI)`

Same problem for maliit-keyboard2.

Hmm I should have been more clear with my issue...
I didn't really know much about CMake then; adding the include directories globaly was a workaround, looks like the wayland include dirs can be solved by adding `Wayland::Client` to `target_link_libraries` for `inputpanel-shell`, but for xkbcommon looks like it is not getting the result at all.
Apparently, some openSUSE packagers intentionally messed with the include directories for pkgconfig to make sure packages are reading them properly, so this is probably why it is only happening until now.

The build error is:
```
[    7s] /usr/include/qt5/QtXkbCommonSupport/5.15.2/QtXkbCommonSupport/private/qxkbcommon_p.h:59:10: fatal error: xkbcommon/xkbcommon.h: No such file or directory
[    7s]    59 | #include <xkbcommon/xkbcommon.h>
```
Inspecting the directory I see the header is there.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,8 +54,8 @@ if(enable-wayland)
     find_package(Wayland REQUIRED)
     find_package(Qt5WaylandClient 5.14 REQUIRED PRIVATE)
     find_package(Qt5XkbCommonSupport REQUIRED PRIVATE)
-    pkg_check_modules(XKBCOMMON REQUIRED xkbcommon)
-    pkg_check_modules(WAYLANDCLIENT REQUIRED wayland-client)
+    pkg_check_modules(XKBCOMMON REQUIRED IMPORTED_TARGET xkbcommon)
+    pkg_check_modules(WAYLANDCLIENT REQUIRED IMPORTED_TARGET wayland-client)
     find_package(Gtk3)
 endif()
 
@@ -110,9 +110,9 @@ qt5_add_dbus_interface(CONNECTION_SOURCES dbus_interfaces/minputmethodserver1int
 add_library(maliit-connection STATIC ${CONNECTION_SOURCES})
 target_link_libraries(maliit-connection Qt5::Core Qt5::DBus Qt5::Gui maliit-common)
 if(enable-wayland)
-    target_link_libraries(maliit-connection Wayland::Client)
+    target_link_libraries(maliit-connection Wayland::Client PkgConfig::XKBCOMMON)
 endif()
-target_include_directories(maliit-connection PUBLIC connection ${XKBCOMMON_INCLUDEDIR})
+target_include_directories(maliit-connection PUBLIC connection)
 
 include_directories(${Qt5Gui_PRIVATE_INCLUDE_DIRS})
 
@@ -310,9 +310,8 @@ if(enable-wayland)
     ecm_add_qtwayland_client_protocol(INPUT_PANEL_SHELL_SOURCES PROTOCOL ${WAYLANDPROTOCOLS_PATH}/unstable/input-method/input-method-unstable-v1.xml BASENAME input-method-unstable-v1)
 
     add_library(inputpanel-shell MODULE ${INPUT_PANEL_SHELL_SOURCES})
-    target_link_libraries(inputpanel-shell Qt5::WaylandClient)
+    target_link_libraries(inputpanel-shell Qt5::WaylandClient PkgConfig::XKBCOMMON PkgConfig::WAYLANDCLIENT)
     target_include_directories(inputpanel-shell PRIVATE ${Qt5WaylandClient_PRIVATE_INCLUDE_DIRS} ${Qt5XkbCommonSupport_PRIVATE_INCLUDE_DIRS})
-    target_include_directories(inputpanel-shell PUBLIC ${WAYLANDCLIENT_INCLUDE_DIRS} ${XKBCOMMON_INCLUDEDIR})
 
     if(enable-wayland-gtk AND TARGET Gtk3::Gtk)
         add_library(im-wayland MODULE

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.3)
+cmake_minimum_required(VERSION 3.6)
 
 project(maliit-framework VERSION 0.99.2)
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,7 +54,8 @@ if(enable-wayland)
     find_package(Wayland REQUIRED)
     find_package(Qt5WaylandClient 5.14 REQUIRED PRIVATE)
     find_package(Qt5XkbCommonSupport REQUIRED PRIVATE)
-
+    pkg_check_modules(XKBCOMMON REQUIRED xkbcommon)
+    pkg_check_modules(WAYLANDCLIENT REQUIRED wayland-client)
     find_package(Gtk3)
 endif()
 
@@ -111,7 +112,7 @@ target_link_libraries(maliit-connection Qt5::Core Qt5::DBus Qt5::Gui maliit-comm
 if(enable-wayland)
     target_link_libraries(maliit-connection Wayland::Client)
 endif()
-target_include_directories(maliit-connection PUBLIC connection)
+target_include_directories(maliit-connection PUBLIC connection ${XKBCOMMON_INCLUDEDIR})
 
 include_directories(${Qt5Gui_PRIVATE_INCLUDE_DIRS})
 
@@ -311,6 +312,7 @@ if(enable-wayland)
     add_library(inputpanel-shell MODULE ${INPUT_PANEL_SHELL_SOURCES})
     target_link_libraries(inputpanel-shell Qt5::WaylandClient)
     target_include_directories(inputpanel-shell PRIVATE ${Qt5WaylandClient_PRIVATE_INCLUDE_DIRS} ${Qt5XkbCommonSupport_PRIVATE_INCLUDE_DIRS})
+    target_include_directories(inputpanel-shell PUBLIC ${WAYLANDCLIENT_INCLUDE_DIRS} ${XKBCOMMON_INCLUDEDIR})
 
     if(enable-wayland-gtk AND TARGET Gtk3::Gtk)
         add_library(im-wayland MODULE

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,7 +55,6 @@ if(enable-wayland)
     find_package(Qt5WaylandClient 5.14 REQUIRED PRIVATE)
     find_package(Qt5XkbCommonSupport REQUIRED PRIVATE)
     pkg_check_modules(XKBCOMMON REQUIRED IMPORTED_TARGET xkbcommon)
-    pkg_check_modules(WAYLANDCLIENT REQUIRED IMPORTED_TARGET wayland-client)
     find_package(Gtk3)
 endif()
 
@@ -310,7 +309,7 @@ if(enable-wayland)
     ecm_add_qtwayland_client_protocol(INPUT_PANEL_SHELL_SOURCES PROTOCOL ${WAYLANDPROTOCOLS_PATH}/unstable/input-method/input-method-unstable-v1.xml BASENAME input-method-unstable-v1)
 
     add_library(inputpanel-shell MODULE ${INPUT_PANEL_SHELL_SOURCES})
-    target_link_libraries(inputpanel-shell Qt5::WaylandClient PkgConfig::XKBCOMMON PkgConfig::WAYLANDCLIENT)
+    target_link_libraries(inputpanel-shell Qt5::WaylandClient PkgConfig::XKBCOMMON Wayland::Client)
     target_include_directories(inputpanel-shell PRIVATE ${Qt5WaylandClient_PRIVATE_INCLUDE_DIRS} ${Qt5XkbCommonSupport_PRIVATE_INCLUDE_DIRS})
 
     if(enable-wayland-gtk AND TARGET Gtk3::Gtk)

